docs/understand/programming_model.rst

```diff
@@ -274,42 +400,65 @@ Global
+
+* **Unified memory**: Structured access reduces the overhead of page migrations.
+
+  Unified memory allows the CPU and GPU to share memory seamlessly, but performance depends on access patterns. Unified memory enables automatic page migration between CPU and GPU memory. However, if different threads access different pages, it can lead to expensive page migrations and slow throughput performance. Accessing unified memory in a structured, warp-friendly manner reduces unnecessary page transfers. Ensure threads access memory in a structured, consecutive manner, minimizing page faults. Prefetch data to the GPU before computation by using ``hipMemPrefetchAsync()``. In addition, using small batch transfers as described below, can reduce unexpected page migrations when using unified memory. 
+
+* **Small batch transfers**: Enable pipelining and improve PCIe bandwidth use.
+
+  Memory transfers between the host and the device can become a major bottleneck if not optimized. One method is to use small batch memory transfers where data is transferred in smaller chunks instead of a dealing with large datasets to avoid long blocking operations. Small batch transfers offer better PCIe bandwidth utilization over large data transfers. Small batch transfers offer performance improvement by offering reduced latency with small batches that run asynchronously using ``hipMemcpyAsync()`` as described in :ref:`asynchronous_how-to`, pipelining data transfers and kernel execution using separate streams. Finally, using pinned memory with small batch transfers enables faster DMA transfers without CPU involvement, greatly improving memory transfer performance. 
+
+Execution model
+===============
 
-  The HIP does not present two *separate* APIs link NVIDIA CUDA. HIP only extends
-  the HIP runtime API with new APIs for ``hipModule`` and ``hipCtx``.
+As previously discussed in :ref:`heterogeneous_programming`, HIP programs consist of two distinct scopes:
+
+* The host-side API running on the host processor. 
+* The device-side kernels running on GPUs. 
+
+Both the host and the device-side APIs have synchronous and asynchronous functions.
 
 Host-side execution
 -------------------
 
-The part of the host-side API which deals with device management and their
-queries are synchronous. All asynchronous APIs, such as kernel execution, data
-movement and potentially data allocation/freeing all happen in the context of
-device streams.
+The host-side API dealing with device management and their queries are synchronous.
+All asynchronous APIs, such as kernel execution, data movement and potentially data
+allocation/freeing all happen in the context of device streams.
 
 Streams are FIFO buffers of commands to execute relating to a given device.
```

Comment:
Please add a link to streams here:
https://rocm.docs.amd.com/projects/HIP/en/latest/how-to/hip_runtime_api/asynchronous.html#streams-and-concurrent-execution

Reply:
Added link